internal/restic/snapshot_find.go

```diff
@@ -38,7 +38,12 @@ func FindLatestSnapshot(ctx context.Context, repo Repository, targets []string,
 		if err != nil {
 			return errors.Errorf("Error loading snapshot %v: %v", snapshotID.Str(), err)
 		}
-		if snapshot.Time.Before(latest) || (hostname != "" && hostname != snapshot.Hostname) {
+
+		if snapshot.Time.Before(latest) {
+			return nil
+		}
+
+		if hostnames != nil && !snapshot.HasHostname(hostnames) {
```

Comment:
Good name for the function! If `HasHostname()` has defined behavior for `hostnames` being an empty (or nil) slice the code here becomes even better :)

Reply:
I already thought about this and dismissed the idea for not being perfect consistent to what the name implies.. :smile: 
However, looking at `HasTagList()` there is already the same functionality. So I added this case, pointed it out in the description and also extended the description of `HasTagList()`. Now it is consistent that `Has"Object"` checks if one of the objects given in the list is satisfied and returns true for an empty list and `Has"Objects"` checks if all objects given in the list is satsified. IMO this makes sense.